signal/playerctl/playerctl_lib.lua

```diff
@@ -230,102 +400,131 @@ local function player_compare_name(name_a, name_b)
-local function player_compare(a, b)
-    local player_a = Playerctl.Player(a)
-    local player_b = Playerctl.Player(b)
-    return player_compare_name(player_a.player_name, player_b.player_name)
+local function player_compare(self, a, b)
+    local player_a = self._private.lgi_Playerctl.Player(a)
+    local player_b = self._private.lgi_Playerctl.Player(b)
+    return player_compare_name(self, player_a.player_name, player_b.player_name)
+end
+
+local function get_current_player_info(self, player)
+    local title = self._private.lgi_Playerctl.Player.get_title(player) or ""
+    local artist = self._private.lgi_Playerctl.Player.get_artist(player) or ""
+    local artUrl = self._private.lgi_Playerctl.Player.print_metadata_prop(player, "mpris:artUrl") or ""
+    local album = self._private.lgi_Playerctl.Player.get_album(player) or ""
+
+    emit_metadata_signal(self, title, artist, artUrl, player.player_name, album, false)
+    playback_status_cb(self, player, player.playback_status)
+    volume_cb(self, player, player.volume)
+    loop_status_cb(self, player, player.loop_status)
+    shuffle_cb(self, player, player.shuffle)
 end
 
-local function start_manager()
-    manager = Playerctl.PlayerManager()
-    if #priority > 0 then
-        manager:set_sort_func(player_compare)
+local function start_manager(self)
+    self._private.manager = self._private.lgi_Playerctl.PlayerManager()
+
+    if #self.priority > 0 then
+        self._private.manager:set_sort_func(function(a, b)
+            return player_compare(self, a, b)
+        end)
     end
 
     -- Timer to update track position at specified interval
-    position_timer = gears.timer {
-        timeout = interval,
-        callback = position_cb
+    self._private.position_timer = gtimer {
+        timeout = self.interval,
+        callback = function()
+            position_cb(self)
+        end,
     }
 
     -- Manage existing players on startup
-    for _, name in ipairs(manager.player_names) do
-        init_player(name)
+    for _, name in ipairs(self._private.manager.player_names) do
+        init_player(self, name)
+    end
+
+    if self._private.manager.players[1] then
+        get_current_player_info(self, self._private.manager.players[1])
     end
 
+    local _self = self
+
     -- Callback to manage new players
-    function manager:on_name_appeared(name)
-        init_player(name)
+    function self._private.manager:on_name_appeared(name)
+        init_player(_self, name)
     end
 
     -- Callback to check if all players have exited
-    function manager:on_name_vanished(name)
-        if #manager.players == 0 then
-            metadata_timer:stop()
-            position_timer:stop()
-            awesome.emit_signal("bling::playerctl::no_players")
+    function self._private.manager:on_name_vanished(name)
+        if #self._private.manager.players == 0 then
+            self._private.metadata_timer:stop()
+            self._private.position_timer:stop()
```

Comment:
These need to be `_self` right?
```suggestion
        if #_self._private.manager.players == 0 then
            _self._private.metadata_timer:stop()
            _self._private.position_timer:stop()
```

Reply:
Not sure how I missed that
cd1a01c